File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
from collections import defaultdict

def solution(id_list, report, k):
    answer = []
    complain = defaultdict(int)
    id_per_complain = defaultdict(list)
    for r in set(report):
        a,b = r.split()
        complain[b] += 1
        id_per_complain[a].append(b)
    complain = set(key for key,val in complain.items() if val >= int(k))
    for id in id_list:
        tmp = complain & set(id_per_complain[id])
        answer.append(len(tmp))
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.swift
import Foundation

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    
    var answer = Array(repeating: 0, count: id_list.count)
    var indexDict: [String: Int] = [:]
    for i in 0..<id_list.count {
        indexDict[id_list[i]] = i
    }
    
    var reportDict: [String: Set<String>] = [:]
    for r in report {
        let arr = r.split(separator: " ").map { String($0) }
        reportDict[arr[1], default:[]].insert(arr[0])
    }
    
    let stopped = reportDict.filter {$0.value.count >= k }
    for stop in stopped.values {
        for s in stop {
            answer[indexDict[s]!] += 1
        }
    }
    
    return answer
}